### DISCUSSION TITLE
Destructuring the store and type inference

### DISCUSSION THREAD

**Discussion by Liwoj** [vuejs/pinia#406] [Help and Questions] [answered]
1st I must say BIG thanks for creating this project. It has soo much better API than Vuex !

My question is about destructuring stores. In case I want to "drill down" into the store and just pass a single property to my component , I do this:

```js
const { user } = toRefs(store)
```

The code works perfectly BUT VS Code + Vetur somehow does not like it. When I hover over `user` variable it is typed as `Ref<unknown>`. Now I just don't know if this is a problem with Pinia typing (probably not as it otherwise works pretty well), typing of `toRefs` or Vetur bug. Im using Vue 2 + @vue/composition-api

Just want to be sure this is correct way how to "split" the store ....

THX

**Comment by posva** [comment, marked as answer]
Doing

```js
const user = toRef(store, 'user')
// or
const { user } = toRefs(store)
```

is correct but it seems like only the first one is correctly inferred while the second is not. This seems to be a bug on Vue types:

```ts
// this is the current version
declare function toRefs<T extends Data = Data>(obj: T): ToRefs<T>;
// this one works
declare function toRefs<T = Data>(obj: T): ToRefs<T>;
```

There might be an open PR already

**Reply by posva** [reply]
Opened https://github.com/vuejs/composition-api/pull/668

**Comment by posva** [comment]
Also, I'm happy you enjoy using Pinia!
